### PR TITLE
Tests: added $request_uri tests with proxy

### PR DESCRIPTION
**Pull request by andrey-zelenkov**

### CODE REVIEW

File: test/test_variables.py [hunk 1]
```diff
@@ -108,6 +108,51 @@ def check_request_uri(req_uri):
...
+            "access_log": {
+                "path": f'{temp_dir}/access.log',
+                "format": "$remote_addr $uri $request_uri",
+            },
+        }
+    )
+
+    assert search_in_file(r'::1', 'access.log') is None
+
+    assert client.get(url='/blah%2Fblah?a=b')['status'] == 200
```
Comment (hongzhidao) [inline test/test_variables.py:146]: Looks good.
But I can't find this type of test for normal proxy. It's useful to add such uri test without "rewrite", for example:
```
"action": {
    "proxy": "http://[::1]:8081",
}

assert client.get(url='/blah%2Fblah?a=b')['status'] == 200

# search /blah%2Fblah?a=b
assert (
        wait_for_record(fr'...', 'access.log') is not None
    ), 'req 8081 (proxy)'
```

**Comment by hongzhidao** [comment]
About the title, there is no `$rewrite_uri` variable in Unit.

File: test/test_variables.py [hunk 2]
```diff
@@ -108,6 +108,51 @@ def check_request_uri(req_uri):
...
+        {
+            "listeners": {
+                "*:8080": {"pass": "routes/a"},
+                "[::1]:8081": {"pass": "routes/b"},
+            },
+            "routes": {
+                "a": [
+                    {
+                        "action": {
+                            "rewrite": "/foo",
```
Comment (hongzhidao) [inline test/test_variables.py:123]: I feel it's necessary to test URI like `/foo%2Ffoo. For example:
```
"rewrite": "/foo%2Ffoo",
```

Comment (andrey-zelenkov) [reply test/test_variables.py]: Added

**Comment by andrey-zelenkov** [comment]
Patch reworked after @hongzhidao's comment and renamed. Commit message updated accordingly.

**Review by hongzhidao — APPROVED** [review]

**Pull request merged**